src/store-api/src/region_request.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use datatypes::arrow::row::Rows;
+
+use crate::metadata::ColumnMetadata;
+use crate::storage::{AlterRequest, ColumnId, ScanRequest};
+
+#[derive(Debug)]
+pub enum RegionRequest {
+    Write(RegionWriteRequest),
+    Read(RegionReadRequest),
+    Delete(RegionDeleteRequest),
+    Create(RegionCreateRequest),
+    Drop(RegionDropRequest),
+    Open(RegionOpenRequest),
+    Close(RegionCloseRequest),
+    Alter(RegionAlterRequest),
+    Flush(RegionFlushRequest),
+    Compact(RegionCompactRequest),
+}
+
+/// Request to write a region.
+#[derive(Debug)]
+pub struct RegionWriteRequest {
+    /// Rows to write.
+    pub rows: Rows,
+    /// Map column name to column index in `rows`.
+    pub name_to_index: HashMap<String, usize>,
```

Comment:
Better to make name_to_index private and generated in `new` to ensure it is correct.

Reply:
Remove it for now. We can add it back later when implement utils for those utils, if necessary